src/InstrumentationEngine.Lib/InstrumentationMethod.cpp

```diff
@@ -88,13 +88,8 @@ HRESULT MicrosoftInstrumentationEngine::CInstrumentationMethod::InitializeCore(
     }
 
     WCHAR wszModuleFullPath[MAX_PATH];
-    memset(wszModuleFullPath, 0, MAX_PATH);
-    if (!PathCanonicalize(wszModuleFullPath, m_bstrModuleFolder))
-    {
-        DWORD dwLastError = GetLastError();
-        CLogging::LogError(_T("CInstrumentationMethod::Initialize - unable to canonicalize method configuration folder: '%s', PID: %u"), m_bstrModuleFolder.m_str, GetCurrentProcessId());
-        return HRESULT_FROM_WIN32(dwLastError);
-    }
+    ZeroMemory(wszModuleFullPath, MAX_PATH * sizeof(WCHAR));
+    wcscpy_s(wszModuleFullPath, MAX_PATH, m_bstrModuleFolder);
 
     hr = StringUtils::SafePathAppend(wszModuleFullPath, m_bstrModule, MAX_PATH);
```

Comment:
>wszModuleFullPath [](http://example.com/codeflow?start=37&length=17)

Without the canonicalization, we run a greater risk of running out of space in wszModuleFuillPath. If m_bstrModuleFolder isn't already canonicalized, then you can have paths like"

"C:\somePath\someSubPath\..\adifferentSubapth\someotherPathHere\,,\,,\realSubPath\\\"

When what you will want is:

"C:\somePath\realSubPath"

If only we could use c++17

Reply:
Note: on Windows, you can use PathCanonicalize, and on Linux, you can use "realpath"

Reply:
You would have to convert back-and-forth from Windows strings, though.

Reply:
I would suggest we fix the canonicalization problem later. The behavior in the PR at this time is the behavior before the Win7 support change and we didn't run into the problems in prior versions.

While I agree that in general this can be a problem, we haven't run into it in practice and I don't think any customer has reported it.

I think we should get the Linux support restored (with the change as-is) and later make more of an effort to fix these types of scenarios.

Reply:
Agreed